MediaBrowser.Providers/TV/SeriesMetadataService.cs

```diff
@@ -177,37 +195,51 @@ private void DeleteEpisode(Episode episode)
         }
 
         /// <summary>
-        /// Creates seasons for all episodes that aren't in a season folder.
+        /// Creates seasons for all episodes if they don't exist.
         /// If no season number can be determined, a dummy season will be created.
+        /// Updates seasons names.
         /// </summary>
         /// <param name="series">The series.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The async task.</returns>
-        private async Task FillInMissingSeasonsAsync(Series series, CancellationToken cancellationToken)
+        private async Task UpdateAndCreateSeasonsAsync(Series series, CancellationToken cancellationToken)
         {
             var seriesChildren = series.GetRecursiveChildren(i => i is Episode || i is Season);
-            var episodesInSeriesFolder = seriesChildren
+            var uniqueSeasonNumbers = seriesChildren
                 .OfType<Episode>()
-                .Where(i => !i.IsInSeasonFolder);
-
-            List<Season> seasons = seriesChildren.OfType<Season>().ToList();
+                .Select(e => e.ParentIndexNumber)
+                .Distinct()
+                .Select(s => s >= 0 ? s : null);
 
-            // Loop through the unique season numbers
-            foreach (var episode in episodesInSeriesFolder)
+            if (uniqueSeasonNumbers is not null)
```

Comment:
It can never be null?

Reply:
it can be null if the LINQ statement returns nothing. It does not throw anywhere within the function though.

Reply:
But uniqueSeasonNumbers is a LINQ statement. How can it be null?

Reply:
According to VSCode `uniqueSeasonNumbers` is of type `IEnumerable<int?>?`. Maybe it is possible if there is no child with `OfType<Episode>()`? Or should the latter result in an empty `IEnumerable`?

Reply:
Whenever you use `var`, the type is nullable.

Reply:
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/declarations#implicitly-typed-local-variables